frontend/src/components/grantaccess/GrantDelegatedAdmin.vue

```diff
@@ -0,0 +1,185 @@
+import {
+    selectedApplicationId,
+    selectedApplicationDisplayText,
+} from '@/store/ApplicationState';
+import { UserType } from 'fam-app-acsctl-api';
+import type { FamRoleDto } from 'fam-admin-mgmt-api/model';
+import { formValidationSchema } from '@/services/utils';
+
+const defaultFormData = {
+    domain: UserType.I,
+    userId: '',
+    verifiedForestClients: [],
+    roleId: null as number | null,
+};
+const formData = ref(JSON.parse(JSON.stringify(defaultFormData))); // clone default input
+
+const delegatedRoleOptions = computed(() => {
+    return LoginUserState.getCachedAppRoles(selectedApplicationId.value!);
+});
+
+/* ------------------ User information method ------------------------- */
+const userDomainChange = (selectedDomain: string) => {
+    formData.value.domain = selectedDomain;
+    formData.value.userId = '';
+};
+
+const userIdChange = (userId: string) => {
+    formData.value.userId = userId;
+};
+
+const verifyUserIdPassed = ref(false);
+const setVerifyUserIdPassed = (verifiedResult: boolean) => {
+    verifyUserIdPassed.value = verifiedResult;
+};
+
+/* ------------------- Role selection method -------------------------- */
+const getSelectedRole = (): FamRoleDto | undefined => {
+    return delegatedRoleOptions?.value.find(
+        (item) => item.id === formData.value.roleId
+    );
+};
+
+const isAbstractRoleSelected = () => {
+    return getSelectedRole()?.type_code == 'A';
+};
+
+const roleSelectChange = (roleId: number) => {
+    formData.value.roleId = roleId;
+};
+
+/* ----------------- Forest client number method ----------------------- */
+const setVerifiedForestClients = (verifiedForestClientNumber: string) => {
+    formData.value.verifiedForestClients.push(verifiedForestClientNumber);
+};
+
+const removeVerifiedForestClients = (index: number) => {
+    formData.value.verifiedForestClients.splice(index, 1);
+};
+
+const resetVerifiedForestClients = () => {
+    formData.value.verifiedForestClients = [];
+};
+
+/* ---------------------- Form method ---------------------------------- */
+const cancelForm = () => {
+    formData.value = defaultFormData;
+    router.push('/dashboard');
+};
+
+/*
+Two verifications are cunnretly in place: userId and forestClientNumber.
+*/
+const areVerificationsPassed = () => {
+    return (
+        // userId verification
+        verifyUserIdPassed.value &&
+        // forestClientNumber verification
+        (!isAbstractRoleSelected() ||
+            (isAbstractRoleSelected() &&
+                formData.value.verifiedForestClients.length > 0))
+    );
+};
+
+const handleSubmit = async () => {
+    // This will be implemented in task #1160
+    console.log('Data to send to backend', formData.value);
+
+    router.push('/dashboard');
+};
+</script>
+
+<template>
+    <PageTitle
+        title="Add a delegated admin"
+        :subtitle="`Adding a delegated admin to ${selectedApplicationDisplayText}. All fields are mandatory`"
```

Comment:
Adjusted this subtitle a little bit, added the name of the selected application, so it's clear to user.

Reply:
Thanks, that makes sense.